src/Microsoft.SqlTools.ServiceLayer/TableDesigner/TableDesignerService.cs

```diff
@@ -1802,6 +1805,9 @@ private Dac.TableDesigner CreateTableDesigner(TableInfo tableInfo)
                 connectionStringBuilder.InitialCatalog = tableInfo.Database;
                 connectionStringBuilder.ApplicationName = ConnectionService.GetApplicationNameWithFeature(connectionStringBuilder.ApplicationName, TableDesignerService.TableDesignerApplicationNameSuffix);
                 var connectionString = connectionStringBuilder.ToString();
+
+                // Sanity checks, passing these checks doesn't guarantee that the user can open the table designer.
+                CheckPermissions(tableInfo, connectionString);
```

Comment:
unfortunately, this is going to have negative impact on the loading time, I am wondering whether we can do the check only after an exception is caught, so that the happy path won't be impacted.

Reply:
makes sense, I've moved the permission check into the catch part